plugins/sentinelone/komand_sentinelone/actions/run_remote_script/action.py

```diff
@@ -0,0 +1,69 @@
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import PluginException
+from .schema import RunRemoteScriptInput, RunRemoteScriptOutput, Input, Output, Component
+
+# Custom imports below
+
+
+class RunRemoteScript(insightconnect_plugin_runtime.Action):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="run_remote_script",
+            description=Component.DESCRIPTION,
+            input=RunRemoteScriptInput(),
+            output=RunRemoteScriptOutput(),
+        )
+
+    def run(self, params={}):
+        agent_ids = params.get(Input.IDS)
+        script_id = params.get(Input.SCRIPT_ID)
+        if not script_id:
+            raise PluginException(
+                cause=f"No script id provided to execute.",
+                assistance=f"Please select a script to execute from the SentinelOne console.",
+            )
+        task_description = params.get(Input.TASK_DESCRIPTION)
+        if not task_description:
+            raise PluginException(
+                cause=f"No task description provided.",
+                assistance=f"Please provide a task description for the script.",
+            )
+
+        script_timeout = params.get(Input.TIMEOUT, 3600)
+        if script_timeout < 60 or script_timeout > 172800:
+            # Timeout outside of allowed parameters - use default value instead
+            script_timeout = 3600
+
+        input_params = params.get(Input.INPUT_PARAMETERS, "")
+        password = params.get(Input.PASSWORD, "")
+        if password:
+            if len(password) <= 10 or " " in password:
+                raise PluginException(
+                    cause="Invalid password.",
+                    assistance="Password must have more than 10 characters and cannot contain whitespace.",
+                )
```

Comment:
I think it's repeated in the action above. Could you please move `10` as a constant variable? Also, it would be great if you could create a separate method for the password check, so as not to duplicate the code 😄

Reply:
I've moved this out to a helper function to avoid duplication.